include/mitsuba/core/distr_1d_cond.h

```diff
@@ -0,0 +1,1225 @@
+};
+
+template <typename Value>
+std::ostream &operator<<(std::ostream &os,
+                         const ConditionalIrregular1D<Value> &distr) {
+    os << "ConditionalIrregular1D[" << std::endl;
+    os << "  enable_sampling = " << string::indent(distr.m_enable_sampling)
+       << std::endl;
+    os << "]";
+    return os;
+}
+
+/**
+ *
+ * \brief Conditional 1D regular distribution
+ *
+ *
+ * Similar to the regular 1D distribution, but this class represents an
+ * N-Dimensional regular one (with the extra conditional dimensions being also
+ * regular).
+ *
+ * As an example, assume you have a 3D distribution P(x,y,z), with leading
+ * dimension X. This class would allow you to obtain the linear interpolated
+ * value of the PDF for \c x given \c y and \c z. Additionally, it allows you to
+ * sample from the distribution P(x|Y=y,Z=z) for a given \c y and \c z.
+ *
+ * It assumes every conditioned PDF has the same size.
+ *
+ * This distribution can be used in the context of spectral rendering, where
+ * each wavelength conditions the underlying distribution.
+ *
+ */
+template <typename Value> class ConditionalRegular1D {
+    using Float        = std::conditional_t<dr::is_static_array_v<Value>,
+                                            dr::value_t<Value>, Value>;
+    using FloatStorage = DynamicBuffer<Float>;
+    using Index        = dr::uint32_array_t<Value>;
+    using UInt32       = dr::uint32_array_t<Float>;
+    using IndexFloat   = dr::uint32_array_t<Float>;
+    using Mask         = dr::mask_t<Value>;
+    using TensorXf     = dr::Tensor<FloatStorage>;
+
+    using ScalarFloat    = dr::scalar_t<Float>;
+    using ScalarVector2f = Vector<ScalarFloat, 2>;
+    using ScalarVector2u = Vector<uint32_t, 2>;
+    using ScalarUInt32   = dr::uint32_array_t<ScalarFloat>;
+
+    using Vector2f = Vector<Float, 2>;
+
+public:
+    ConditionalRegular1D() : m_enable_sampling(true) {};
+
+    /**
+     * \brief Construct a conditional regular 1D distribution
+     *
+     * \param pdf
+     *  Flattened array of shape [D1, D2, ..., Dn, N] containing the PDFs
+     * \param range
+     *  Range where the leading dimension N is defined
+     * \param range_cond
+     *  Array of ranges where the dimensional conditionals are defined
+     * \param size_cond
+     *  Array with the size of each conditional dimension
+     * \param enable_sampling
+     *  Enable sampling of the distribution
+     * \param nearest
+     *  If true, the sampling will return the nearest neighbor
+     */
+    ConditionalRegular1D(const FloatStorage &pdf, const ScalarVector2f &range,
+                         const std::vector<ScalarVector2f> &range_cond,
+                         const std::vector<ScalarUInt32> &size_cond,
+                         bool enable_sampling = true, bool nearest = false)
+        : m_range(range), m_size_cond(size_cond), m_range_cond(range_cond),
+          m_enable_sampling(enable_sampling), m_nearest(nearest) {
```

Comment:
I think what I've written for the irregular distribution's `enable_samping` and `nearest`, also applies here.